src/DataFramesMeta.jl

```diff
@@ -139,6 +141,16 @@ end
 
 fun_to_vec(kw::QuoteNode) = kw
 
+function repair_source(x)
+    if isempty(x) || isconcretetype(eltype(x))
+        return x
+    elseif all(t -> t isa Union{AbstractString, Symbol}, x)
+        return [xi isa AbstractString ? Symbol(xi) : xi for xi in x]
```

Comment:
I think this is better for two reasons:
1. `Symbol(:x)` is no-op so no need for `if`
2. if `x` is not a vector then comprehension will convert it to a vector (but maybe you want it? How do you e.g. handle `Tuple`? Or maybe we should check that we got a vector?)

```suggestion
        return Symbol.(x)
```

Reply:
Thanks. 

w.r.t vector inputs, I'm using a `Vector` of column references in `@where` even though it's not strictly necessary. This is the right thing to do to avoid too much compilation.